File: src/gestorAplicacion/usuario/Colchon.java
package gestorAplicacion.usuario;

import gestorAplicacion.economia.Divisa;

import java.time.LocalDate;

public class Colchon extends Cuenta{
	
	/**
	 * 
	 */
	private static final long serialVersionUID = 514183129515203955L;

	private LocalDate fechaRetiro;


	public Colchon(Usuario usuario, Divisa divisa, String nombre, LocalDate fechaRetiro) {
		super(usuario,divisa, nombre);
		setUsuario(usuario);
		setFechaRetiro(fechaRetiro);
	}

	public boolean retirar(double monto){
		if(this.fechaRetiro.isBefore(LocalDate.now())){
			return super.retirar(monto);
		}
		return false;


	}
	public LocalDate getFechaRetiro() {
		return fechaRetiro;
	}

	public void setFechaRetiro(LocalDate fechaRetiro) {
		this.fechaRetiro = fechaRetiro;
	}
}

File: Python/gestorAplicacion/usuario/Colchon.py
from gestorAplicacion.usuario import Cuenta
from datetime import date, time, datetime

class Colchon(Cuenta):

    def __init__(self, usuario, divisa, nombre, fechaRetiro):
        
        self._fechaRetiro = None

        super().__init__(usuario,divisa, nombre)
        self.setUsuario(usuario)
        self.setFechaRetiro(fechaRetiro)

    def retirar(self, monto):
        if self._fechaRetiro.isBefore(date.today()):
            return super().retirar(monto)
        return False


    def getFechaRetiro(self):
        return self._fechaRetiro

    def setFechaRetiro(self, fechaRetiro):
        self._fechaRetiro = fechaRetiro
